pandas/core/generic.py

```diff
@@ -2458,8 +2462,22 @@ def to_msgpack(self, path_or_buf=None, encoding='utf-8', **kwargs):
         return packers.to_msgpack(path_or_buf, self, encoding=encoding,
                                   **kwargs)
 
-    def to_sql(self, name, con, schema=None, if_exists='fail', index=True,
-               index_label=None, chunksize=None, dtype=None, method=None):
+    # TODO: Replace `con: Any` and `Callable[[Any, Any, ...` when SQLTable
+    # and sqlalchemy can be imported. SQLTable can't be imported due to
```

Comment:
you don’t need all of these comments
just use a quote around the type

you don’t need to import them

Reply:
Thanks for the hint. @WillAyd this may be good to have in #26869 .

Reply:
Just an FYI adding quoted paths here doesn't really do anything - reveal_type still shows these as Any:

```sh
pandas/core/generic.py:2591: error: Revealed type is 'Any'
```

Forward references this like this I think only work within a module and not cross module, as shown by Mypy documentation:

https://mypy.readthedocs.io/en/latest/kinds_of_types.html#class-name-forward-references

The only reason this works is because we ignore all missing imports in our mypy config, so an actual error here with the string name is suppressed and replaced with Any. This might not hold true if we go through with #26645

Very nuanced but I think better if we DIDN'T do these types of annotations in the code base

Reply:
I got the correct type for `sqlite3` when I tried `reveal_type()`. @WillAyd did you notice the `if TYPE_CHECKING:` imports? It seems I'm making a mistake with sqlalchemy though, since it's revealed as `Any` as you mentioned.

```
pandas\core\generic.py:2619: error: Revealed type is 'Union[Any, sqlite3.dbapi2.Connection]'
pandas\core\generic.py:2620: error: Revealed type is 'Union[builtins.str, def (pandas.io.sql.SQLTable, Union[Any, sqlite3.dbapi2.Connection], builtins.list[builtins.str], typing.Iterator[builtins.list[Any]]), None]'
```

I'll figure `sqlalchemy` types out and make sure this passes the tests out.

Also, here is the cross-module example: https://mypy.readthedocs.io/en/latest/common_issues.html#import-cycles